### PR TITLE
Fixes in `ImpliedDoLoop`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
This PR solves  [#981 (comment) ](https://github.com/lfortran/lfortran/issues/981#issuecomment-1304725752), AST->ASR passes for ImpliedDoLoop works.

### CODE REVIEW

**Comment by certik** [comment]
Did you copy & pasted the code from body visitor? If so, then just put it into the common visitor, so that we do not duplicate code.

**Comment by Pranavchiku** [comment]
Yeah, okay.

**Comment by Pranavchiku** [comment]
@certik this PR can have a final review.

File: src/lfortran/semantics/ast_symboltable_visitor.cpp [hunk 1]
```diff
@@ -509,6 +509,7 @@ class SymbolTableVisitor : public CommonVisitor<SymbolTableVisitor> {
         return r;
     }
     
+
```
Comment (certik) [inline src/lfortran/semantics/ast_symboltable_visitor.cpp:512]:
```suggestion
```

**Review by certik — APPROVED** [review]
I think this looks good now, thanks!

**Pull request merged**